package/src/package/section.rs

```diff
@@ -0,0 +1,267 @@
+use alloc::{
+    borrow::{Cow, ToOwned},
+    format,
+    string::ToString,
+};
+use core::{fmt, str::FromStr};
+use miden_assembly_syntax::DisplayHex;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+/// A unique identifier/tag for optional sections of the Miden package format
+///
+/// The following tag ranges are reserved for specific categories of use:
+///
+/// * `0` - unused
+/// * `1..10` - reserved for non-rollup use cases, e.g. debug information
+/// * `10..100` - reserved for rollup use cases, e.g. account component metadata
+/// * `100..254` - reserved for TBD use cases
+/// * `255` - assigned to custom user-defined sections, which must also be uniquely named
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[repr(u8)]
+pub enum SectionId {
+    //--- CORE SECTIONS
+    /// The section containing debug information (source locations, spans)
+    DebugInfo = 1,
+
+    //--- ROLLUP SECTIONS
+    /// This section provides the encoded metadata for a compiled account component
+    ///
+    /// Currently, this corresponds to the serialized representation of
+    /// `miden-objects::account::AccountComponentMetadata`, i.e. name, descrioption, storage, that
+    /// is associated with this package.
+    AccountComponentMetadata = 10,
```

Comment:
It feels a bit weird that here in `miden-vm` we need to know about how custom sections may be used downstream in `miden-objects`. Ideally, `miden-vm` creates should not be aware of `miden-objects` or anything else in `miden-base`.

Could we not use the `Custom` section for these? That is, what is the main thing we gain from making `AccountComponentMetadata` special here?

Reply:
The idea behind non-`Custom` variants here is to represent "well-known" custom sections, which are important in more than one place in the toolchain. By assigning a well-known section identifier to those, the encoding is more compact, and it is implied we can reason about the format of the data in that section (i.e. the section isn't completely opaque).

We don't _have_ to do that with account component metadata, but since that was previously the only actual custom section we were emitting anywhere (i.e. from the compiler), it seems a bit odd to treat it as completely opaque to packages in general (since any account package will effectively require it).